assets/in

```diff
@@ -136,12 +136,19 @@ if [ "$submodules" != "none" ]; then
 
     submodules="$(git config --list --name-only | sed -ne 's/^submodule\.\(.*\)\.url$/\1/p' | sort -u)"
   else
-    submodules=$(echo $submodules | jq -r '(.[])')
+    submodules=$(
+      echo $submodules | jq -r '.[]' | while read path; do
+        git config --file .gitmodules --name-only --get-regexp '\.path$' $path |
+          sed -e 's/^submodule\.\(.\+\)\.path$/\1/'
+      done
+    )
 
     git submodule init $submodules
```

Comment:
According to the documentation, `git submodule init` also expects paths (and not names), so i think this is wrong as `$submodules` is the list of _names_ of the submodules here.

I think it would make things simpler and cleaner if we would work with submodule _paths_ everywhere, as that's what is passed in via the `submodule` parameter and that's what the `git submodule` sub-commands expect. Then we could avoid the extra steps to convert them to names and then converting the names back to paths. It would be then a single change to line 137 plus the test.

Reply:
I tried this for a long time but for reasons I could not understand it kept breaking the submodule deepening test. It seemed like the update hook that we set in the config wasn't being used at all. 

Speaking of which, I believe we need the name for configuring that hook.